staging/src/github.com/kubeedge/mapper-framework/hack/make-rules/generate.sh

```diff
@@ -11,10 +11,14 @@ MAPPER_DIR="$(cd "$(dirname "$ROOT_DIR")" && pwd -P)"
 
 function entry() {
   # copy template
-  read -p "Please input the mapper name (like 'Bluetooth', 'BLE'): " -r mapperName
-  if [[ -z "${mapperName}" ]]; then
-    echo "the mapper name is required"
-    exit 1
+  if [ $# -eq 0 ] ;then
```

Comment:
You may write it this way: 
```
  mapperName=$1
  if [[ -z $mapperName ]]; then
    read -p "Please input the mapper name (like 'Bluetooth', 'BLE'): " -r mapperName
    if [[ -z "${mapperName}" ]]; then
      echo "the mapper name is required"
      exit 1
    fi
  fi
```

You can also use this to remind the user to enter their name:
```
  mapperName=$1
  while [[ -z $mapperName  ]];do
    read -p "Please input the mapper name (like 'Bluetooth', 'BLE'): " -r mapperName
  done
  echo $mapperName
```

Reply:
I tried running the code you gave, but an error occurred

Reply:
![generate sh failed](https://github.com/kubeedge/kubeedge/assets/48212997/cf1fb431-cbd8-4a3e-9d4b-6ff9fa98bf72)